### ISSUE TITLE
Cant make texture from byte array

### ISSUE THREAD

**Issue by IrishBruse** [NogginBops/ImGui.NET_OpenTK_Sample#10]
I am having trouble making a texture for use with Imgui.image from a byte array with r g b im new to using opentk so any help would be appreciated thanks.
also im using the latest 4.0 opentk

**Comment by NogginBops** [comment]
Could you give some more details on what it is you want to do?
From what it sounds like this would be more of a ImGui.NET question.

**Comment by IrishBruse** [comment]
Im making an emulator and my window data is in a byte array ordered rgb and im trying to use the texture class you provide to pass the image data to the imgui window for displaying basically render the current frame to an imgui window.
![image](https://user-images.githubusercontent.com/14964651/92402046-d1417280-f126-11ea-9dc8-17e0bd7b6bdc.png)
![image](https://user-images.githubusercontent.com/14964651/92402122-f1713180-f126-11ea-8805-0e9340111e1c.png)
This is what gets rendered a black box even though it should be noise.
![image](https://user-images.githubusercontent.com/14964651/92402227-19609500-f127-11ea-9ebf-e468f1211ed5.png)

**Comment by NogginBops** [comment]
Ah ok now I understand what it is that you are trying to do. 

What is happening here is a combination of multiple things I believe.

First thing is that the internal format for textures using that ctor is either `Srgb8Alpha8` or `Rgba8` depending on the `srgb` boolean flag. So either you need to provide alpha data or modify the `Texture` class to be able to handle a `Rgb8` format.
https://github.com/NogginBops/ImGui.NET_OpenTK_Sample/blob/master/Dear%20ImGui%20Sample/Texture.cs#L107
(it's worth noting that the GPU will probably store the data as `Rgba8` anyways, so just adding the alpha component is probably the best solution)

Next is that because the method was originally meant for `Bitmap` data pointers the `format` is defined as `Bgra` because that is how the [endian-ness of modern machines works out](https://stackoverflow.com/questions/8104461/pixelformat-format32bppargb-seems-to-have-wrong-byte-order). So if you have your data as bytes directly with the in memory byte order of `Red-byte`, `Green-byte`, `Blue-byte` then you want to just specify the format as `Rgb`
https://github.com/NogginBops/ImGui.NET_OpenTK_Sample/blob/master/Dear%20ImGui%20Sample/Texture.cs#L113
(and once again note that you probably want to specify alpha too so it would be `Argb`)

Please try these modifications and tell me how it went 🙂

**Comment by IrishBruse** [comment]
So I think this would be the changes needed to the ctor for it to work correctly.
![image](https://user-images.githubusercontent.com/14964651/92406655-7ca2f500-f130-11ea-93b0-8d1c76fc9666.png)
I think its really strange as I would expect something to happen like the texture to be weird colors or an error.

I added the alpha component to the test array so I would expect to see random pixel transparent with the randomize but its all still black with no error.
![image](https://user-images.githubusercontent.com/14964651/92406438-f25a9100-f12f-11ea-8cfe-a1f6afce5803.png)

**Comment by NogginBops** [comment]
Can you show the code you use for displaying the image using ImGui?
Or if you could zip your project and send it to me so I can test it on my machine.

**Comment by IrishBruse** [comment]
https://github.com/IrishBruse/Monoboy

**Comment by IrishBruse** [comment]
![image](https://user-images.githubusercontent.com/14964651/92407091-9264ea00-f131-11ea-95cb-aa8c0a674714.png)
Also thats the imgui drawing

**Comment by NogginBops** [comment]
I figured out the issue you are having.
It's the same issue as #9 .
How I figured that out is by looking at the `framebufferTexture.GLTexture` value in a debugger and noting that it's `0`.
The fix is to before calling any OpenTK/OpenGL functions to call `MakeCurrent()`.

Adding a `MakeCurrent()` here fixes the problem.
```cs
MakeCurrent();
framebufferTexture = new Texture("FrameBuffer", Constant.WindowWidth, Constant.WindowHeight, dataPRT);
```

This is a OpenTK 4-pre9.4 issue and calling `MakeCurrent()` will likely not be necessary in future OpenTK versions.
